x-pack/plugin/relevance-search/src/main/java/org/elasticsearch/xpack/relevancesearch/query/RelevanceMatchQueryBuilder.java

```diff
@@ -104,45 +116,47 @@ protected void doXContent(final XContentBuilder builder, final Params params) th
         if (relevanceSettingsId != null) {
             builder.field(RELEVANCE_SETTINGS_FIELD.getPreferredName(), relevanceSettingsId);
         }
-
+        if (curationsSettingsId != null) {
+            builder.field(CURATIONS_SETTINGS_FIELD.getPreferredName(), curationsSettingsId);
+        }
+        boostAndQueryNameToXContent(builder);
         builder.endObject();
     }
 
+    public void setRelevanceSettingsId(String relevanceSettingsId) {
+        this.relevanceSettingsId = relevanceSettingsId;
+    }
+
     @Override
-    protected Query doToQuery(final SearchExecutionContext context) throws IOException {
+    protected void doWriteTo(final StreamOutput out) throws IOException {
+        out.writeString(query);
+        out.writeOptionalString(relevanceSettingsId);
+        out.writeOptionalString(curationsSettingsId);
+    }
 
-        Map<String, Float> fieldsAndBoosts;
-        if (relevanceSettingsId != null) {
-            try {
-                RelevanceSettings relevanceSettings = relevanceSettingsService.getRelevanceSettings(relevanceSettingsId);
-                QueryConfiguration queryConfiguration = relevanceSettings.getQueryConfiguration();
-                fieldsAndBoosts = queryConfiguration.getFieldsAndBoosts();
-            } catch (RelevanceSettingsService.RelevanceSettingsNotFoundException e) {
-                throw new IllegalArgumentException("[relevance_match] query can't find search settings: " + relevanceSettingsId);
-            } catch (RelevanceSettingsService.RelevanceSettingsInvalidException e) {
-                throw new IllegalArgumentException("[relevance_match] invalid relevance search settings for: " + relevanceSettingsId);
-            }
-        } else {
-            Collection<String> fields = queryFieldsResolver.getQueryFields(context);
-            if (fields.isEmpty()) {
-                throw new IllegalArgumentException("[relevance_match] query cannot find text fields in the index");
-            }
-            fieldsAndBoosts = fields.stream().collect(Collectors.toMap(Function.identity(), (field) -> AbstractQueryBuilder.DEFAULT_BOOST));
-        }
+    String getRelevanceSettingsId() {
+        return relevanceSettingsId;
+    }
 
-        final CombinedFieldsQueryBuilder builder = new CombinedFieldsQueryBuilder(query, fieldsAndBoosts);
+    @Override
+    protected Query doToQuery(final SearchExecutionContext context) throws IOException {
+        return queryRewriter.rewriteQuery(this, context);
```

Comment:
QueryRewriter is used for this, to separate the parsing / serialization / deserialization form the actual query building process